bowbuffer.go

```diff
@@ -34,37 +39,34 @@ func NewBuffer(size int, typ Type) Buffer {
 	return buf
 }
 
-// Len returns the length of the Buffer
 func (b Buffer) Len() int {
-	switch data := b.Data.(type) {
-	case []int64:
-		return len(data)
-	case []float64:
-		return len(data)
-	case []bool:
-		return len(data)
-	case []string:
-		return len(data)
+	switch b.DataType {
+	case Int64:
+		return len(b.Data.([]int64))
+	case Float64:
+		return len(b.Data.([]float64))
+	case Boolean:
+		return len(b.Data.([]bool))
+	case String:
+		return len(b.Data.([]string))
 	default:
-		panic(fmt.Errorf("unsupported type '%T'", b.Data))
+		panic(fmt.Errorf("unsupported type '%s'", b.DataType))
 	}
 }
 
-// SetOrDrop sets the value `value` at index `i` by attempting a type conversion to the Buffer Type.
-// Set the bit in the Buffer nullBitmapBytes if the conversion succeeded, or clear it otherwise.
 func (b *Buffer) SetOrDrop(i int, value interface{}) {
 	var valid bool
-	switch v := b.Data.(type) {
-	case []int64:
-		v[i], valid = Int64.Convert(value).(int64)
-	case []float64:
-		v[i], valid = Float64.Convert(value).(float64)
-	case []bool:
-		v[i], valid = Boolean.Convert(value).(bool)
-	case []string:
-		v[i], valid = String.Convert(value).(string)
+	switch b.DataType {
+	case Int64:
+		b.Data.([]int64)[i], valid = Int64.Convert(value).(int64)
```

Comment:
Can we consider a refactoring to remove type assertions which are increasing the technical debt and code base complexity ? 

https://medium.com/extreme-programming/avoiding-type-assertions-in-go-6feaa87

Reply:
I thought really hard about it already, and couldn't find a better solution. How would you do it?
I get a 404 not found on the link you sent :(

Reply:
Issue created

Reply:
There are ways to avoid type assertion like making a dedicated buffer struct for each type and use an interface externally. Historically I choose to expose internal data to let users implement logics around their need instead of having non extensible buffer as interfaces. They are made for mutability after all. 
But after some time more methods were added like SetOrDrop etc and it might not be relevant anymore to expose internal data. Again all this have to be rethanked with generic features in mind. It's a game changer.

Reply:
Yes, we need to start experimenting with the new generics, it is promising :)